src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -263,12 +275,143 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
 
         if(invalidFields.length > 0){
             let fieldListAsString = invalidFields.join(', ');
-            showToast('Invalid Form', 'The following fields are required: ' + fieldListAsString, 'error');
+            this.hasPageLevelError = true;
+            this.pageLevelErrorMessageList = [ {index: 0, errorMessage: `The following fields are required: ${fieldListAsString}`} ];
         }
 
         return invalidFields.length === 0;
     }
 
+    /**
+     * validates donation donor type on sectionsList
+     * @param sectionsList, list of sections
+     * @returns {boolean|*} - true if form invalid, false otherwise
+     */
+    isDonorTypeInvalid( sectionsList ){
+
+        const DONATION_VALUES = [
+            DONATION_DONOR_FIELDS.donationDonorField,
+            DONATION_DONOR_FIELDS.account1ImportedField, DONATION_DONOR_FIELDS.account1NameField,
+            DONATION_DONOR_FIELDS.contact1ImportedField, DONATION_DONOR_FIELDS.contact1LastNameField
+        ];
+        // get label and value using apiName as key from fields for each section
+        let miniFieldWrapper = {};
+        sectionsList.forEach( section => {
+            miniFieldWrapper = { ...miniFieldWrapper, ...(section.getFieldValueAndLabel(DONATION_VALUES)) };
+        });
+
+        // if no donation donor selection, nothing to validate here yet
+        if ( isEmpty(miniFieldWrapper[DONATION_DONOR_FIELDS.donationDonorField].value) ) {
+            return false;
+        }
+        
+        // is donation donor valid
+        return this.getDonorTypeValidationError( miniFieldWrapper, sectionsList );
```

Comment:
Does this function return a Boolean, or does it return an error message? Based on the definition and doc header it should always return a Boolean.

Reply:
returns boolean but before doing that displays the appropriate error (if any)